components/LessonCard.tsx

```diff
@@ -9,11 +9,16 @@ type Props = {
   lessonCount: number
   hourCount: string | number
   description: string
+  complete?: boolean
+  inProgress?: boolean
```

Comment:
lessons cannot be `complete` and `inProgress` at the same time. Doing this could be confusing later on. I think `currentState?: string` is a good one.

This current combination gives us 4 states: `true/true`, `false/false`, `true/false`, `false/true`

We really need 3 states, so having a `string` to represent state is better imo.
`inProgress`, `completed`, or default.

Reply:
This will make your `containerClass` logic easier to manager down the road.